proposals/1533-host-specific-tasklist.md

```diff
@@ -0,0 +1,241 @@
+
+#### Example
+
+The basic usage looks like the following (it belongs to a larger workflow):
+
+```go
+sessionCtx, err := CreateSession(ctx)
+
+if err != nil {
+
+    // Creation failed. Wrong ctx or too many outstanding sessions.
+
+}
+
+err = ExecuteActivity(sessionCtx, DownloadFileActivity, filename).Get(sessionCtx, nil)
+
+if err != nil {
+
+    // Session(worker) has failed or activity itself returns an error.
+
+    // User can perform normal error handling here and decide whether creating a new session is needed.
+
+    // If they decide to create a new session, they need to call CompleteSession() so that worker resource can be released. It’s recommended that users create a function for session and call defer CompleteSession(sessionCtx) after a session is created. If session retry is needed, this function can be called multiple times if the error returned is retriable.
+
+}
+
+err = ExecuteActivity(sessionCtx, ProcessFileActivity, filename).Get(sessionCtx, nil)
+
+if err != nil {
+
+    // Session(worker) has failed or activity itself returns an error.
+
+}
+
+err = ExecuteActivity(sessionCtx, UploadFileActivity, filename).Get(sessionCtx, nil)
+
+if err != nil {
+
+    // Session(worker) has failed or activity itself returns an error.
+
+}
+
+err = CompleteSession(sessionCtx)
+
+if err != nil {
+
+    // Wrong ctx is used or failed to release session resource.
+
+}
+```
+
+#### New Worker Options
+
+There will be three new options available when creating a worker:
+
+* **EnableSessionWorker**
+
+  The flag controls whether the worker will accept activities belongs to a session.
+
+* **SessionResourceID**
+
+  An identifier of the resource that will be consumed if a session is executed on the worker. For example, if a worker has a GPU, and a session for training a machine learning model is executed on the worker, then some memory on the GPU will be consumed. In this case, the resourceID can be the identifier of the GPU.
+
+**NOTE: user need to ensure there’s only one worker using a certain resourceID.**
+
+* **MaxCurrentSessionExecutionSize**
+
+  Since a session may consume some kind of resource, user can use this option to control the maximum number of sessions running in the worker process at the same time.
+
+## Implementation
+
+### Session Model
+
+All sessions are **resource specific**. It basically means a session is always tied to some kind of resource (not the worker). The resource can be CPU, GPU, memory, file descriptors etc and we want all the activities within a session to be executed by the same worker which owns the resource. If worker dies and restarts, as long as it owns the same resource, we can try to reestablish the session (this is future work). Also user needs to ensure that resource is owned by only one worker.
```

Comment:
It basically means -> This means that
file descriptors etc and -> file descriptors, etc., and
the same worker which owns -> the same worker that owns
If worker dies -> If a worker dies
Also user -> Also, the user
that resource is -> that the resource is